### PR TITLE
5521 Add rake script to migrate old documents from legacy system

### PR DESCRIPTION
**Pull request by X0nic**

### CODE REVIEW

File: lib/tasks/migrate_data.rake [hunk 1]
```diff
@@ -27,5 +27,54 @@ namespace :tww do
...
 
+    documents_path = File.join root_path, 'documents'
+    documents = Dir.entries(documents_path)
+      .reject { |f| File.directory?(f) || f[0].include?('.') }
+      .map do |doc|
+        arr = doc.split('.')
+        Document.new(File.join(documents_path, doc), doc, arr[0], arr[1], :document)
+      end
+
+    contracts_path = File.join root_path, 'contracts'
```
Comment (X0nic) [inline lib/tasks/migrate_data.rake]: There is some duplication, but for a one time use script is it a problem?

Comment (hooverlunch) [inline lib/tasks/migrate_data.rake]: Nope!

File: lib/tasks/migrate_data.rake [hunk 2]
```diff
@@ -27,5 +27,54 @@ namespace :tww do
...
+    root_path = '/tmp/madeline'
+    Document = Struct.new(:full_path, :file_name, :document_name, :loan_id, :document_kind)
 
+    documents_path = File.join root_path, 'documents'
+    documents = Dir.entries(documents_path)
+      .reject { |f| File.directory?(f) || f[0].include?('.') }
+      .map do |doc|
+        arr = doc.split('.')
+        Document.new(File.join(documents_path, doc), doc, arr[0], arr[1], :document)
+      end
```
Comment (hooverlunch) [inline lib/tasks/migrate_data.rake]: Very nice!

File: lib/tasks/migrate_data.rake [hunk 3]
```diff
@@ -27,5 +27,54 @@ namespace :tww do
...
+          uploaded_file_name = file.file_name.tr(' ', '_') # Carrierwave will replace spaces with underscores
+          if loan.media.exists?(item: uploaded_file_name)
+            print '*'
+          else
+            print '.'
+            loan.media.create(item: File.open(file.full_path), kind_value: file.document_kind)
+          end
+        end
+        loan.save!
+      rescue
```
Comment (hooverlunch) [inline lib/tasks/migrate_data.rake]: Please catch a the ActiveRecord::RecordNotFound exception explicitly. Else could get some very confusing bugs here.

File: lib/tasks/migrate_data.rake [hunk 4]
```diff
@@ -27,5 +27,54 @@ namespace :tww do
...
+        loan = Loan.find key
+
+        value.each do |file|
+          uploaded_file_name = file.file_name.tr(' ', '_') # Carrierwave will replace spaces with underscores
+          if loan.media.exists?(item: uploaded_file_name)
+            print '*'
+          else
+            print '.'
+            loan.media.create(item: File.open(file.full_path), kind_value: file.document_kind)
+          end
```
Comment (hooverlunch) [inline lib/tasks/migrate_data.rake]: Fantastic! I'm assuming you've tested the idempotence here?

Comment (X0nic) [inline lib/tasks/migrate_data.rake]: Sure did. Even works when some files are loaded, and some are not.

**Pull request merged**